linchpin/provision/roles/aws/tasks/provision_aws_ec2_vpc_nat_gateway.yml

```diff
@@ -0,0 +1,33 @@
+---
+- name: "Provision/Teardown AWS VPC NAT Gateways"
+  ec2_vpc_nat_gateway:
+    name: "{{ res_def['name'] }}"
```

Comment:
Does it need a name field?

Reply:
yes, because youc could have number of nat gateways in AWS.

Reply:
But nobody uses this "name", and nobody would return this. Why would you still have it? And you can use the gateway_id to differentiate them. Even you really need a name, should we return this somewhere? Like appending this after the topo_output?

Reply:
the name should be list in the resources file (not the easiest to understand file but non the less)
From user perspective when troubleshooting, if there is no control over the name, then one must go to the resources file, find the right ID and then look for it AWS. I understand your point, and 99% of the time probably no one will use this option, but what is the harm in it?

Reply:
@14rcole , @samvarankashyap what do you think?

Reply:
NM, the name field doesn't exist in the specification. I don't know how it slipped into my mind it does. removing.

Reply:
FYI, I removed it from the provisionin task, but schema requires name feilds.